portal/src/main/webapp/js/src/oncoprint/studyview.js

```diff
@@ -5,10 +5,96 @@ $(document).ready(function() {
 		interpolate: /\{\{(.+?)\}\}/g
 	};
 	$('#oncoprint_controls').html(_.template($('#main-controls-template').html())());
+        
+        var clinical_data_promises = []
+        var GenePanelData = [];
+        var gene_panel_clinical_coll = new ClinicalColl();
+        clinical_data_promises.push(
+        gene_panel_clinical_coll.fetch({
+        type: "POST",
+        data: {
+           cancer_study_id: cancer_study_id_selected,
+           attribute_id: "GENE_PANEL",
+           case_list: cases
+        },
+        success: function(response) {
+               GenePanelData = response.toJSON();
+           }
+        }));
+        
+        //fetch genepanel
+        var gene_panel_files = [{path:"api/genepanel/IMPACT341", file:"IMPACT341"},{path:"api/genepanel/IMPACT410",file:"IMPACT410"}];
+        var genePanel = [];
+        var gainGenepanel = function(filename,datafetchArray){
+            datafetchArray.push(
+            $.getJSON(filename,function(result){
+                genePanel[(filename.split("/"))[2]] = result;
+            }));
+        }
+        
+        
+        for(var i=0; i<gene_panel_files.length; i++)
+        {
+            var gene_panel_file = gene_panel_files[i];
+            var gene_panel_file_name = gene_panel_files.file;
+            clinical_data_promises.push($.getJSON(gene_panel_file.path, function(result) { gene_panel_file_data[gene_panel_file_name] = result; }));
```

Comment:
I made a mistake, sorry

because of async issues we need to use a closure here
The code inside this loop should just be

```var gene_panel_file = gene_panel_files[i];
clinical_data_promises.push(
              $.getJSON(gene_panel_file.path, (function(filename) {
                                                                          return function(result) { gene_panel_file_data[filename] = result; }; })(gene_panel_file.file)));```